src/extension.ts

```diff
@@ -86,25 +107,52 @@ const parseFile = async (source: string) => {
     return;
   }
 
-  const index = nodes.findIndex((node) => node.path === source);
+  const index = nodes.findIndex((node) => node.path === filePath);
   if (index !== -1) {
     nodes[index].label = title;
   } else {
-    nodes.push({ id: id(source), path: source, label: title });
+    nodes.push({ id: id(filePath), path: filePath, label: title });
   }
 
-  edges = edges.filter((edge) => edge.source !== id(source));
+  // remove edges based on an old version of this file
+  edges = edges.filter((edge) => edge.source !== filePath);
+
   const links = findLinks(ast);
 
   for (const link of links) {
-    const parentDirectory = source.split("/").slice(0, -1).join("/");
-    const target = path.normalize(`${parentDirectory}/${link}`);
+    let target = link;
+    if (!path.isAbsolute(link)) {
+      const parentDirectory = filePath.split("/").slice(0, -1).join("/");
```

Comment:
Now when I look at it, this line might be moved out of the `for` loop, right?

Reply:
Correct. This is what happens when I push the results of `git merge` without giving them more attention

Reply:
Oh, nevermind, that's not what you were talking about. Anyways, yes, I will move it.